packages/elements/src/dialog/index.ts

```diff
@@ -145,6 +148,18 @@ export class Dialog extends Overlay {
 
   public noCancelOnOutsideClick = true;
 
+
+  /**
+   * Called once after the component is first rendered
+   * @param changedProperties map of changed properties with old values
+   * @returns {void}
+   */
+  protected firstUpdated (changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+
+    this.setAttribute('aria-modal', String(!this.noInteractionLock));
```

Comment:
Only on first update ?

Reply:
After I reconsider on this point, I think this should move to `updated` and condition to `shouldUpdate`